code/modules/awaymissions/corpse.dm

```diff
@@ -96,8 +104,8 @@
 		if(ishuman(M))
 			var/mob/living/carbon/human/hoomie = M
 			hoomie.body_type = mob_gender
-	if(faction)
-		M.faction = list(faction)
+	if(LAZYLEN(faction))
+		M.faction = faction
```

Comment:
```suggestion
	M.faction = faction
```

Reply:
This would change behavior though - mob faction would get nulled if there are no factions on the mob spawner instead of them retaining their default ones listed on /mob.